src/main/kotlin/no/nav/tiltakspenger/skjerming/SkjermingService.kt

```diff
@@ -34,30 +36,77 @@ class SkjermingService(
+            keyValue("behovId", packet["@behovId"].asText())
+        )
+        sikkerlogg.info(
+            "løser behov med id {} og korrelasjonsid {}",
+            keyValue("id", packet["@id"].asText()),
+            keyValue("behovId", packet["@behovId"].asText())
+        )
+        sikkerlogg.debug { "mottok melding: ${packet.toJson()}" }
+        sikkerlogg.debug { "mottok melding: ${packet.toJson()}" }
+    }
+
+    private fun loggVedUtgang(packet: JsonMessage, løsning: () -> String) {
+        logg.info(
+            "har løst behov med id {} og korrelasjonsid {}",
+            keyValue("id", packet["@id"].asText()),
+            keyValue("behovId", packet["@behovId"].asText())
+        )
+        sikkerlogg.info(
+            "har løst behov med id {} og korrelasjonsid {}",
+            keyValue("id", packet["@id"].asText()),
+            keyValue("behovId", packet["@behovId"].asText())
+        )
+        sikkerlogg.info { "publiserer løsning: $løsning" }
+    }
+
+    private fun loggVedFeil(ex: Throwable, packet: JsonMessage) {
+        logg.error(
+            "feil: ${ex.message} ved behandling av behov {}",
+            keyValue("id", packet["@id"].asText()),
+            ex
+        )
+        sikkerlogg.error(
+            "feil: ${ex.message} ved behandling av behov {}",
+            keyValue("id", packet["@id"].asText()),
+            ex
+        )
+    }
+
     override fun onPacket(packet: JsonMessage, context: MessageContext) {
-        LOG.info { "Received packet: ${packet.toJson()}" }
-        val ident = packet["ident"].asText()
-        val behovId = packet["@behovId"].asText()
-
-        val erSkjermet = runBlocking {
-            skjermingKlient.erSkjermetPerson(
-                fødselsnummer = ident,
-                behovId = behovId,
+        runCatching {
+            loggVedInngang(packet)
+            val ident = packet["ident"].asText()
+            val behovId = packet["@behovId"].asText()
+            sikkerlogg.debug { "mottok ident $ident" }
+
+            val erSkjermet = runBlocking {
+                skjermingKlient.erSkjermetPerson(
+                    fødselsnummer = ident,
+                    behovId = behovId,
+                )
+            }
+
+            packet["@løsning"] = mapOf(
+                BEHOV.SKJERMING to erSkjermet
             )
-        }
-        LOG.debug { "Received ident $ident" }
-        packet["@løsning"] = mapOf(
-            BEHOV.SKJERMING to erSkjermet
-        )
-        LOG.info { "Sending skjermet: $erSkjermet" }
-        context.publish(packet.toJson())
+            loggVedUtgang(packet) { "$erSkjermet" }
+            context.publish(packet.toJson())
+        }.onFailure {
+            loggVedFeil(it, packet)
+        }.getOrThrow()
     }
 
     override fun onError(problems: MessageProblems, context: MessageContext) {
-        LOG.debug { problems }
+        logg.info { "meldingen validerte ikke: $problems" }
     }
 
+    @Suppress("EmptyFunctionBlock")
```

Comment:
kan man ikke bare ta den bort da?

Reply:
Joda, i og for seg. Jeg tenkte bare det var greit å vite at den er der? :)